pkg/reconciler/apiresource/controller.go

```diff
@@ -21,56 +21,59 @@ import (
 	"time"
 
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
-	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	typedapiextensions "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
-	crdexternalversions "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
+	crdinfomer "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions/apiextensions/v1"
 	crdlister "k8s.io/apiextensions-apiserver/pkg/client/listers/apiextensions/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog"
 
 	apiresourcev1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apiresource/v1alpha1"
-	kcpclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
 	typedapiresource "github.com/kcp-dev/kcp/pkg/client/clientset/versioned/typed/apiresource/v1alpha1"
-	kcpexternalversions "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	apiresourceinformer "github.com/kcp-dev/kcp/pkg/client/informers/externalversions/apiresource/v1alpha1"
 	apiresourcelister "github.com/kcp-dev/kcp/pkg/client/listers/apiresource/v1alpha1"
 	"github.com/kcp-dev/kcp/pkg/util/errors"
 )
 
-const resyncPeriod = 10 * time.Hour
 const clusterNameAndGVRIndexName = "clusterNameAndGVR"
 
 func GetClusterNameAndGVRIndexKey(clusterName string, gvr metav1.GroupVersionResource) string {
 	return clusterName + "$" + gvr.String()
 }
 
-func NewController(cfg *rest.Config, autoPublishNegotiatedAPIResource bool) (*Controller, error) {
-	apiresourceClient := typedapiresource.NewForConfigOrDie(cfg)
+func NewController(
+	apiResourceClient typedapiresource.ApiresourceV1alpha1Interface,
```

Comment:
If we keep passing in the `*rest.Config` it would help my refactor be much smaller, since cluster-aware clientsets do not exist in your version of `kubernetes`, but they do in mine, and we need to use those in the controller

Reply:
Will adjust to passing in a clientset instead of typed clients. Large discussion here: https://kubernetes.slack.com/archives/C021U8WSAFK/p1636461810203200. Can adjust later if needed.